src/org/usfirst/frc/team2399/robot/OI.java

```diff
@@ -130,4 +96,24 @@ public OI(){
 		shiftToHotButt.whenPressed(shiftToHot);
 		shiftToDangerousButt.whenPressed(shiftToDangerous);
 	}
+	
+	public static double getLeftY() {
+		return leftJoy.getY() * RobotMap.JOYDRIVE_FORWARD;
+	}
+
+	public static double getRightY() {
+		return rightJoy.getY() * RobotMap.JOYDRIVE_FORWARD;
+	}
+	
+	public static double deadband(double input)
+	{
+		if(Math.abs(input) < RobotMap.DEADBAND)
+		{
+			return 0;
+		}
+		else
+		{
+			return ((Math.abs(input)/input) *((Math.abs(input)-RobotMap.DEADBAND)/(1-RobotMap.DEADBAND)));
```

Comment:
This is a bit difficult to understand. Such is the nature of fitting equations into code; however, I might suggest one improvement for readability: Since you call `Math.abs(input)` multiple times, consider creating a variable and storing the result of `Math.abs(input)` in it once, then using the variable in the equation. A well-named variable might make it more readable, and therefore more maintainable. If you have a strong argument to the contrary, I will definitely consider it. Feel free to refactor the equation in other ways as well.